lib/babel.js

```diff
@@ -74,8 +77,33 @@ if (inputSourceMap) {
 	options.inputSourceMap = JSON.parse(inputSourceMap.map);
 }
 
+// load up custom compilers
+process.argv.slice(3).forEach(function (arg) {
+	var match = /^--compilers=.+?:(.+)$/i.exec(arg);
+	if (match) {
+		require(match[1]);
+	}
+});
+
 // include test file
-var transpiled = babel.transformFileSync(testPath, options);
+var Module = module.constructor;
+var source = fs.readFileSync(testPath, 'utf8');
+var extension = path.extname(testPath);
+
+var compiler = Module._extensions[extension];
+if (compiler) {
+	compiler(
+		{
+			_compile: function (s, f) {
+				source = s;
+				testPath = f;
```

Comment:
[Proxyquire does](https://github.com/thlorenz/proxyquire/blob/b9b47a48901059a411f0f88870009db9f03f02d2/lib/proxyquire.js#L241) but that wouldn't impact the test file.



> I'm leery of mocking an undocumented API this way.

[nyc mocks it](https://github.com/bcoe/nyc/blob/17dd92eb358b3c8e0fb4abdb34228bbbadcb689b/index.js#L151) but ignores the second argument to `_compile()`.